core/app/service/setting.go

```diff
@@ -517,17 +466,3 @@ func checkCertValid() error {
 
 	return nil
 }
-
-func loadPort(address string) string {
-	re := regexp.MustCompile(`(?:(?:\[([0-9a-fA-F:]+)\])|([^:/\s]+))(?::(\d+))?$`)
-	matches := re.FindStringSubmatch(address)
-	if len(matches) <= 0 {
-		return ""
-	}
-	var port string
-	port = matches[3]
-	if len(port) != 0 {
-		return port
-	}
-	return ""
-}
```

Comment:
There do not seem to be any specific changes or discrepancies mentioned in terms of code quality, complexity, performance, security vulnerabilities, etc. However, there are some common conventions such as removing unused imports at the top level, organizing imports appropriately according to Go's naming convention, and ensuring consistent use of functions within their own class or module rather than outside it.

For instance:

```
func getSomeValue() int { ... } // put inside this function instead of being out side
``` 

The code is generally well-structured but might benefit slightly from more explicit checks like using defer statements when you need to perform tasks that will continue after the block ends. This helps ensure those cleanup operations happen before the code executes. Other small refinements include adding spaces around operators and comments where appropriate, capitalizing variable names consistently, ensuring meaningful parameter lists for loops and other sections involving iteration. In the context of Golang standard library modules, `import` directives should ideally follow conventional import style.

In summary, there doesn't appear to be major defects or areas requiring modifications except possibly improvements in documentation about usage, clarity, readability, and consistency which could greatly enhance future maintenance efforts.